pkg/plugins/gateway/lambda/lambda.go

```diff
@@ -253,15 +255,30 @@ func (s *LambdaGateway) handle(ctx context.Context, data map[string]interface{})
 			}
 		}
 	}
+
 	return nil, nil
 }
 
 // Start the lambda gateway handler
 func (s *LambdaGateway) Start(pool worker.WorkerPool) error {
 	// s.finished = make(chan int)
-	s.pool = pool
+
 	// Here we want to begin polling lambda for incoming requests...
-	s.runtime(s.handle)
+	s.runtime(func(ctx context.Context, data map[string]interface{}) (interface{}, error) {
+		s.pool = &worker.InstrumentedWorkerPool{
+			WorkerPool: pool,
+			Wrapper:    worker.InstrumentedWorkerFn(spanFromContext(ctx), false, true),
+		}
```

Comment:
should we be re-wrapping this on every request? We could update the handler signatures on the base worker pools to accepts context, and add the span creating in the instrumented worker pools handle methods, so we can wrap once for all contexts.